test/unit/math/prim/prob/inv_wishart_cholesky_rng_test.cpp

```diff
@@ -0,0 +1,129 @@
+#include <stan/math/prim.hpp>
+#include <test/unit/math/prim/util.hpp>
+#include <boost/random/mersenne_twister.hpp>
+#include <boost/math/distributions/chi_squared.hpp>
+#include <gtest/gtest.h>
+#include <stdexcept>
+#include <vector>
+
+TEST(ProbDistributionsInvWishartCholesky, rng) {
+  using Eigen::Dynamic;
+  using Eigen::Matrix;
+  using Eigen::MatrixXd;
+
+  using stan::math::inv_wishart_cholesky_rng;
+
+  boost::random::mt19937 rng;
+
+  MatrixXd omega(3, 4);
+  EXPECT_THROW(inv_wishart_cholesky_rng(3.0, omega, rng), std::domain_error);
+
+  MatrixXd sigma(3, 3);
+  sigma << 9.0, -3.0, 0.0, -3.0, 4.0, 1.0, 0.0, 1.0, 3.0;
+
+  Matrix<double, Dynamic, Dynamic> LS = sigma.llt().matrixL();
+
+  EXPECT_NO_THROW(inv_wishart_cholesky_rng(3.0, LS, rng));
+  EXPECT_THROW(inv_wishart_cholesky_rng(2, LS, rng), std::domain_error);
+  EXPECT_THROW(inv_wishart_cholesky_rng(-1, LS, rng), std::domain_error);
+  LS(2, 2) = -1.0;
```

Comment:
You don't need the `.0` everywhere.  The compiler figures out whether constants should be `double` or `int`.  The only exception is arithmetic with them. 

Now for the tests like the following, it makes sense to try both integer and floating point args to make sure the polymorphism is handled correctly.